### PR TITLE
Spark 525339 popover

### PR DESCRIPTION
**Pull request by shnaaz**
# Description

- Added role and aria-modal to ModalContainer
- Added aria-labelledby to ModalConatiner if interactive
- aria-haspopup and id to triggerComponent

# Links
[Tech Breakdown](https://confluence-eng-gpk2.cisco.com/conf/display/WTWC/Bucket+Breakdown+-+SPARK-522085+Overlays)
[SPARK-525339](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-525339)

### CODE REVIEW

File: src/components/ModalContainer/ModalContainer.unit.test.tsx [hunk 1]
```diff
@@ -34,6 +34,26 @@ describe('<ModalContainer />', () => {
       expect(container).toMatchSnapshot();
     });
 
+    it('should match snapshot with role', () => {
+      expect.assertions(1);
+
+      const role = 'dialog';
+
+      const container = mount(<ModalContainer role={role} />);
```
Comment (jor-row) [inline src/components/ModalContainer/ModalContainer.unit.test.tsx:42]: This snapshot test should pass something other than role=dialog (as it's the default and will be in every other snapshot anyway)

File: src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -82,17 +87,35 @@ const Popover = forwardRef((props: Props, ref: ForwardedRef<HTMLElement>) => {
...
+  const handleOnPopoverShow = useCallback(() => {
+    setIsVisible(true);
+  },[]);
+
   useEffect(() => {
     firstFocusElement?.focus();
   }, [firstFocusElement]);
 
+  const triggerComponentCommonProps = {
+    'aria-haspopup': 'dialog',
```
Comment (jor-row) [inline src/components/Popover/Popover.tsx:107]: aria-haspopup can have other values like "listbox" or "menu" which I think we might want to use in other components (e.g. SelectNext might need "listbox"). I think this should be a prop that defaults to 'dialog'

Comment (shnaaz) [reply src/components/Popover/Popover.tsx:107]: As discussed we will check with buttons team if they are handling this.

Comment (jor-row) [reply src/components/Popover/Popover.tsx:107]: Sounds good. Depending on what they say you might not have to remove this prop if interactive=false but regardless, I think aria-haspopup should be the same value as the modalcontainer role (as per those mdn docs we looked at)

**Review by jor-row — APPROVED** [review]
Nice, thanks for making those changes

**Pull request merged**